server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -94,6 +100,29 @@ public RemoteFsTranslog(
         }
     }
 
+    public static void download(TranslogTransferManager translogTransferManager, Path location) throws IOException {
+        TranslogTransferMetadata translogMetadata = translogTransferManager.readMetadata();
+        if (translogMetadata != null) {
+            if (Files.notExists(location)) {
+                Files.createDirectories(location);
+            }
+            // Delete translog files on local before downloading from remote
+            for (Path file : FileSystemUtils.files(location)) {
+                Files.delete(file);
+            }
+            Map<String, String> generationToPrimaryTermMapper = translogMetadata.getGenerationToPrimaryTermMapper();
+            for (long i = translogMetadata.getGeneration(); i >= translogMetadata.getMinTranslogGeneration(); i--) {
```

Comment:
if segments upload is lagging, then just downloading the most recent translog might not be enough. Can we create a tracking issue? If it exists already, pls do share.

Reply:
Already tracking it here: https://github.com/opensearch-project/OpenSearch/issues/3754